tests/test_region_aware.py

```diff
@@ -139,91 +145,125 @@ def test_region_aware_county():
+        Graph,
+        tree,
+    )
     from functools import partial
     import random
 
     graph = Graph.from_json("tests/graphs_for_test/8x8_with_muni.json")
     population_col = "TOTPOP"
 
-    muni_names= [str(i) for i in range(1,17)]
-    county_names = [str(i) for i in range(1,5)]
+    muni_names = [str(i) for i in range(1, 17)]
+    county_names = [str(i) for i in range(1, 5)]
 
     updaters = {
         "population": updaters.Tally(population_col, alias="population"),
         "cut_edges": updaters.cut_edges,
         "splits": updaters.tally_region_splits(["muni", "county"]),
     }
-    initial_partition = Partition(graph, assignment="district", updaters=updaters) 
+    initial_partition = Partition(graph, assignment="district", updaters=updaters)
 
     ideal_pop = sum(initial_partition["population"].values()) / len(initial_partition)
-    weights = {"muni": 0.5, "county": 0.5}
     num_steps = steps
     epsilon = 0.01
 
     random.seed(seed)
-    weighted_proposal = partial(proposals.recom,
-                                pop_col=population_col,
-                                pop_target=ideal_pop,
-                                epsilon=epsilon,
-                                weight_dict=weights,
-                                node_repeats=10,
-                                method=partial(tree.bipartition_tree, max_attempts=10000))
-
-    weighted_chain = MarkovChain(proposal=weighted_proposal,
-                                 constraints=[constraints.contiguous],
-                                 accept=accept.always_accept,
-                                 initial_state=initial_partition,
-                                 total_steps=num_steps)
-    
+    surcharged_proposal = partial(
+        proposals.recom,
+        pop_col=population_col,
+        pop_target=ideal_pop,
+        epsilon=epsilon,
+        region_surcharge=surcharges,
+        node_repeats=10,
+        method=partial(
+            tree.bipartition_tree,
+            max_attempts=10000,
+            warn_attempts=warn_attempts,
+        ),
+    )
+
+    surcharged_chain = MarkovChain(
+        proposal=surcharged_proposal,
+        constraints=[constraints.contiguous],
+        accept=accept.always_accept,
+        initial_state=initial_partition,
+        total_steps=num_steps,
+    )
+
     n_muni_splits = -1
     n_county_splits = -1
-    for item in weighted_chain:
+    for item in surcharged_chain:
         n_muni_splits = item["splits"]["muni"]
         n_county_splits = item["splits"]["county"]
-    
+
     return (n_muni_splits, n_county_splits)
 
+
+def test_region_aware_muni_warning():
+    with pytest.warns(UserWarning) as record:
+        # Random seed 2 should succeed, but drawing the
+        # tree is hard, so we should get a warning
+        run_chain_dual(
+            seed=2,
+            steps=1000,
+            surcharges={"muni": 2.0, "county": 2.0},
+            warn_attempts=2,
+        )
+
+    random.seed(2018)
```

Comment:
why does this random seed need to be set after `run_chain_dual`?

Reply:
Ah, this is a product of my neuroticism. The seed is changed in the `run_chain_dual` function, and when I was running a custom set of tests, I wanted to be sure to reset to the base seed of 2018. This can probably be removed now.